puro/lib/src/env/command.dart

```diff
@@ -145,5 +154,34 @@ Future<int> runDartCommand({
   final exitCode = await dartProcess.exitCode;
   await stdoutFuture;
   await stderrFuture;
+
+  await disposeExitSignals();
+
   return exitCode;
 }
+
+/// Capture SIGINT and SIGTERM signals. If we don't capture them, the parent
+/// process will exit, so the dart command won't have a chance to handle them.
+/// Some CLI apps might want to behave differently when they receive these
+/// signals.
+Future<void> Function() _setupExitSignals(ProcessStartMode mode) {
+  StreamSubscription<ProcessSignal>? sigIntSub, sigTermSub;
+
+  if (mode == ProcessStartMode.normal ||
```

Comment:
Shouldn't this only be for `ProcessStartMode.inheritStdio`? The child process doesn't appear to receive signals for `ProcessStartMode.normal`

Reply:
@pingbird  Testing on Linux.

Doing the following changes:
* Update the `runDartCommand` mode to be .normal
* Remove the `if mode == .normal`. 

If I run the sample program from above, output is not shown (expected due to the .normal mode), but doing Ctrl+C will stop the program at once, instead of needing to send the signal 3 times.

Reply:
I placed the check because that would be the correct behavior if normal mode was used, but I can remove it if you want, since it's not used

Reply:
At least on windows, this condition would completely prevent you from ctrl-c working all together while puro runs pub get on the flutter tool

Reply:
What steps should I take to test what you are describing on Linux? I don't think I'm following you.

Reply:
The behavior is a little different on Linux, the child process still gets sent a SIGINT with `ProcessStartMode.normal` (maybe because it's part of the same process group?) regardless I think the correct behavior is for ctrl-c to quit puro when the user is not actually interacting with a child process, i.e. when the mode is not `ProcessStartMode.inheritStdio`

Reply:
Alright, makes sense. I've just pushed the changes